TinkerPop 3.0 support

amcp opened the pull request:
Upgrade use of traversal sources for TP3, other TP3 syntax
Use orient-gremlin until OrientDB is officially supported
Dependency version upgrades
Other small improvements

sarovios commented:
Do we have any performance benchmarks between TP API and Neo4j native API?

sarovios commented:
This is not related with this PR but I just find it out. I am not able to build the project, it seems that dynamoDBLocal is not on Maven Central as I get this: `No versions available for com.amazonaws:DynamoDBLocal:jar:[1.10, 2.0.0)`. Please add the fix to README.md

amcp commented:
I updated the README.md with some preliminary benchmark results. OrientDB is not officially at TP3 yet so I did not run these tests. I made some interesting findings especially with regards to how we were setting custom vertex ids in Titan. It turns out that our original implementation of custom vertex ids never worked. I found this out by enabling assertions (-ea) with the JVM and saw Titan core crap out in many places. I fixed the issues with setting custom ids in Titan, but I am not sure we stand to gain much from custom ids, so I will play around and see if using auto-ids improves performance.

sarovios commented:
Shall I close that and create a new PR when this is ready?
